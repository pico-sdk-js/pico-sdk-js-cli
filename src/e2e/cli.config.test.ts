import { describe, beforeEach, it } from '@jest/globals';
import psjRunner from './psjRunner';

describe('PSJ Runtime Scenarios', () => {
    describe('.config', () => {
        beforeEach(async () => {
            // Reset pico after each run
            // prettier-ignore
            await psjRunner()
                .start(['--skip-header'])
                .command('.format --confirm')
                .command(`.restart`);
        });

        it('initially has no autorun setting', async () => {
            // prettier-ignore
            await psjRunner()
                .start(['--skip-header'])
                .command('.config autorun')
                .assertSnapshot();
        });

        it('can set an autorun file', async () => {
            const fileText = "print('success!');";

            // prettier-ignore
            await psjRunner()
                .start(['--skip-header'])
                .command(`.write test.js --content "${fileText}"`)
                .command('.config autorun test.js')
                .command(`.restart`)
                .pause(100)
                .assertSnapshot();
        });

        it('can unset an autorun file', async () => {
            const fileText = "print('success!');";

            // prettier-ignore
            await psjRunner()
                .start(['--skip-header'])
                .command(`.write test.js --content "${fileText}"`)
                .command('.config autorun test.js')
                .command(`.restart`)
                .pause(100)
                .command('.config autorun --unset')
                .command(`.restart`)
                .assertSnapshot();
        });
    });
});
