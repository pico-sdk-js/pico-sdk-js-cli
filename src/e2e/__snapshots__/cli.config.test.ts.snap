// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PSJ Runtime Scenarios .config can set an autorun file: stdOut 1`] = `
"Searching for device running Pico-SDK-JS engine...
Connected to Pico-SDK-JS Engine v0.0.3 at '/dev/ttyACM0'
> .write test.js --content "print('success!');"
Writing "static content" to "test.js"
18 bytes (1 segments) written
> .config autorun test.js
Set config option autorun to "test.js"
> .restart
Device restarting
> 
success!
> "
`;

exports[`PSJ Runtime Scenarios .config can unset an autorun file: stdOut 1`] = `
"Searching for device running Pico-SDK-JS engine...
Connected to Pico-SDK-JS Engine v0.0.3 at '/dev/ttyACM0'
> .write test.js --content "print('success!');"
Writing "static content" to "test.js"
18 bytes (1 segments) written
> .config autorun test.js
Set config option autorun to "test.js"
> .restart
Device restarting
> 
success!
> .config autorun --unset
Unset config option autorun
> .restart
Device restarting
> "
`;

exports[`PSJ Runtime Scenarios .config initially has no autorun setting: stdOut 1`] = `
"Searching for device running Pico-SDK-JS engine...
Connected to Pico-SDK-JS Engine v0.0.3 at '/dev/ttyACM0'
> .config autorun
null
> "
`;
